### PR TITLE
Adding control flow guard compiler option for Windows builds

### PR DESCRIPTION
**Pull request by ganeshkamath89**
More about CFG: https://learn.microsoft.com/en-us/windows/win32/secbp/control-flow-guard 
Issue originally identified by BlackDuck Security analysis tool

### CODE REVIEW

File: Configurations/10-main.conf [hunk 1]
```diff
@@ -1475,10 +1475,10 @@ my %targets = (
                                 "UNICODE", "_UNICODE",
                                 "_CRT_SECURE_NO_DEPRECATE",
                                 "_WINSOCK_DEPRECATED_NO_WARNINGS"),
-        lib_cflags       => add("/Zi /Fdossl_static.pdb"),
+        lib_cflags       => add("/guard:cf /Zi /Fdossl_static.pdb"),
         lib_defines      => add("L_ENDIAN"),
-        dso_cflags       => "/Zi /Fddso.pdb",
-        bin_cflags       => "/Zi /Fdapp.pdb",
+        dso_cflags       => "/guard:cf /Zi /Fddso.pdb",
+        bin_cflags       => "/guard:cf /Zi /Fdapp.pdb",
```
Comment (tom-cosgrove-arm) [inline Configurations/10-main.conf:1481]: Is there a minimum Visual Studio version for this option? Will it cause errors when used on older versions?

**Comment by t8m** [comment]
I think this would be OK with CLA: trivial.

File: Configurations/10-main.conf (hunk 1 above)
Comment (levitte) [reply Configurations/10-main.conf:1481]: It goes as far back as VS 2015, at least according to docs:

https://learn.microsoft.com/en-us/cpp/build/reference/guard-enable-control-flow-guard?view=msvc-140

**Comment by ganeshkamath89** [comment]
Is there a way to decide the inclusion of control flow guard based on VS version or msbuild version?

**Comment by tom-cosgrove-arm** [comment]
> Is there a way to decide the inclusion of control flow guard based on VS version or msbuild version?

I don't know - that's why I'm asking these questions...

**Comment by levitte** [comment]
@tom-cosgrove-arm's way with simply asking the compiler would be usable too for perl function that wants to figure things out that way.  The `lib_cflags` attribute would be changed the same way still.

**Comment by tom-cosgrove-arm** [comment]
> > Is there a way to decide the inclusion of control flow guard based on VS version or msbuild version?
>
> Yes it's possible to do this, but requires a bit of perl hackery

At which point this PR goes beyond the scope of `CLA: trivial`, and would require a CLA.

But I do like that (is it okay to assume that `perl Configure` is run after PATH is set for the compiler?)

**Comment by ganeshkamath89** [comment]
I have added the commit using 
CLA: Trivial

**Comment by tom-cosgrove-arm** [comment]
Hi, would you mind rebasing to remove the merge commit?

**Comment by ganeshkamath89** [comment]
> Hi, would you mind rebasing to remove the merge commit?

Hi. Is it okay if I remove this pull request and then create one more pull request?

I don't mind doing what you are asking. I am not able to figure it out by myself.

**Comment by ganeshkamath89** [comment]
> Hi, would you mind rebasing to remove the merge commit?

Hi @tom-cosgrove-arm , I have done git squash as explained in this link: https://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

Can you please check once?

**Comment by tom-cosgrove-arm** [comment]
> Can you please check once?

Err, there are now four commits, with two merges. Not sure I can give you a simple few commits to repair that.

> Hi. Is it okay if I remove this pull request and then create one more pull request?

This would be fine

**Pull request closed without merging**